src/features/smartSearch/components/SmartSearchDialog/SmartSearch.tsx

```diff
@@ -0,0 +1,153 @@
+  const intl = useIntl();
+
+  const isEmptyQuery = !filterArray.length && !readOnly;
+
+  const [selectedFilter, setSelectedFilter] =
+    useState<SelectedSmartSearchFilter>(null);
+
+  const [searchState, setSearchState] = useState(
+    isEmptyQuery ? STATE.START_WITH : STATE.PREVIEW
+  );
+
+  return (
+    <>
+      {searchState === STATE.PREVIEW && (
+        <QueryOverview
+          filters={filterArray} // optional
+          hasSaveCancelButtons={hasSaveCancelButtons}
+          onCloseDialog={onDialogClose} //optional
+          onDeleteFilter={(filter) => {
+            setSearchState(STATE.PREVIEW);
+            deleteFilter(filter.id);
+          }}
+          onEditFilter={(filter) => {
+            setSelectedFilter(filter);
+            setSearchState(STATE.EDIT);
+          }}
+          onOpenFilterGallery={() => setSearchState(STATE.GALLERY)}
+          onOpenStartsWithEditor={() => setSearchState(STATE.START_WITH)}
+          onSaveQuery={() => {
+            if (onSave) {
+              onSave({
+                filter_spec: filters,
+              });
+            }
+          }}
+          readOnly={readOnly}
+          startsWithAll={startsWithAll}
+        />
+      )}
+      {searchState === STATE.GALLERY && (
+        <FilterGallery
+          onAddNewFilter={(type) => {
+            setSelectedFilter({ type });
+            setSearchState(STATE.EDIT);
+          }}
+          onCancelAddNewFilter={() => setSearchState(STATE.PREVIEW)}
+        />
+      )}
+      {searchState === STATE.EDIT && selectedFilter && (
+        <FilterEditor
+          filter={selectedFilter}
+          onCancelSubmitFilter={() => setSearchState(STATE.PREVIEW)}
+          onSubmitFilter={(filter) => {
+            if (onOutputConfigured) {
+              onOutputConfigured([
+                {
+                  config: {
+                    filters: filters,
+                  },
+                  title: intl.formatMessage({
+                    id: 'misc.views.columnDialog.localSmartSearch.title',
+                  }),
+                  type: COLUMN_TYPE.LOCAL_SMART_SEARCH,
+                },
+              ]);
+            }
+
+            setSearchState(STATE.PREVIEW);
+            // If editing existing filter
+            if ('id' in filter) {
+              editFilter(filter.id, filter);
+            } else {
+              // If creating a new filter
+              addFilter(filter);
+            }
+          }}
+        />
+      )}
+      {searchState === STATE.START_WITH && (
+        <StartsWith
+          onCancel={() => setSearchState(STATE.PREVIEW)}
+          onSubmit={(shouldStartWithAll) => {
+            if (onOutputConfigured) {
+              onOutputConfigured([
+                {
+                  config: {
+                    filters: filters,
+                  },
+                  title: intl.formatMessage({
+                    id: 'misc.views.columnDialog.localSmartSearch.title',
```

Comment:
The string path is incorrect, there is a `choices` missing. Also, maybe should use `columnTitle` instead of `title`?

![image](https://user-images.githubusercontent.com/550212/215801782-f8e24ef4-44d0-4da1-8aff-8396e12dfa78.png)

(Note that in the screenshot above, I'm running a local version where I've renamed `filters` to `filter_spec` in order to proceed with my testing. See separate comment.)